File: post-day-4/src/topic_pkg/script/topic_subscriber.py
#!/usr/bin/env python3

import rospy
from topic_pkg.msg import Topic

def callback(data):
    rospy.loginfo("\n\n\033[1;36mMessage Data from \033[1;34mPython: \033[1;36m\n")
    rospy.loginfo("\033[1;0mUsia:\033[1;36m %ld Tahun\n", data.usia)
    rospy.loginfo("\033[1;0mTinggi:\033[1;36m %lfcm\n", data.tinggi)
    rospy.loginfo("\033[1;0mNama:\033[1;36m %s\n", data.nama)
    rospy.loginfo("\033[1;0mKehadiran:\033[1;36m %s\n", "Hadir" if data.kehadiran else "Tidak Hadir")

def listener():
    rospy.init_node("topic_subscriber_py", anonymous=True)
    rospy.Subscriber("topic_pkg_py", Topic, callback)

    rospy.spin()


if __name__ == '__main__':
    try:
        listener()
    except rospy.ROSInterruptException:
        pass

File: post-day-4/src/topic_pkg/src/topic_subscriber.cpp
#include "ros/ros.h"
#include "topic_pkg/Topic.h"

void callbackHandler(const topic_pkg::Topic::ConstPtr &msg) {
    ROS_INFO("\n\n\033[1;36mMessage Data: \n");
    ROS_INFO("\033[1;0mUsia:\033[1;36m %ld Tahun\n", msg->usia);
    ROS_INFO("\033[1;0mTinggi:\033[1;36m %lfcm\n", msg->tinggi);
    ROS_INFO("\033[1;0mNama:\033[1;36m %s\n", msg->nama.c_str());
    ROS_INFO("\033[1;0mKehadiran:\033[1;36m %s\n", msg->kehadiran ? "Hadir" : "Tidak Hadir");
}

int main(int argc, char** argv) {
    ros::init(argc, argv, "topic_subscriber");
    
    ros::NodeHandle nh;
    ros::Subscriber topic_subscriber = nh.subscribe("topic_pkg", 1000, callbackHandler);

    ros::spin();

    return 0;
}